catalyst/utils/parser.py

```diff
@@ -3,13 +3,14 @@
 import copy
 from pathlib import Path
 
-from catalyst.utils.config import load_config
-from catalyst.utils.dict import merge_dicts
+from omegaconf import OmegaConf
 
 
 def parse_config_args(*, config, args, unknown_args):
     """@TODO: Docs. Contribution is welcome."""
     for arg in unknown_args:
+        if not arg.startswith("-"):
+            continue
```

Comment:
why do we need this?

Reply:
Example: `catalyst--dl run --foo=bar:str foo=bar`
`--foo=bar:str` - catalyst-style argument
`foo=bar` - omegaconf-style argument
Both arguments will be processed in this block, but "foo=bar" cannot be parsed with
`value_content, value_type = value.rsplit(":", 1)` line because "foo=bar".rsplit(":", 1) gives 1 value instead 2.
So I have added this condition to separate catalyst-style arguments and omegaconf-style arguments.

Reply:
and what about more complex case like
```
catalyst--dl run --/stages/stage1/optimizer_params/opmizer=MyCustomOptimizer:str
```
or
```
catalyst--dl run --/stages/stage1/optimizer_params/opmizer_staps='[10, 20, 30]':list
```

is it possible with omega-conf? @otherman16 @omry

Reply:
just wondering how omega-conf could resolve different python types